src/main/kotlin/platform/architectury/version/ArchitecturyVersion.kt

```diff
@@ -0,0 +1,144 @@
+                    )
+                ); it
+            }.find { mcVersion >= it[0] && mcVersion < it[1] }!!.first()
+        return versions[roundedVersion]!!.asSequence()
+            .sortedDescending()
+            .take(50)
+            .toList()
+    }
+
+    companion object {
+        private fun findMcVersion(architecturyVersion: SemanticVersion): SemanticVersion {
+            val meta = Json.parseToJsonElement(
+                URL(
+                    "https://gist.githubusercontent.com" +
+                        "/shedaniel/4a37f350a6e49545347cb798dbfa72b3" +
+                        "/raw/architectury.json"
+                ).readText()
+            ).jsonObject
+            return SemanticVersion.parse(
+                meta["versions"]!!.jsonObject.filter {
+                    it.value.jsonObject["api"]!!.jsonObject["filter"]!!.jsonPrimitive.content.toRegex().matches(
+                        architecturyVersion.toString()
+                    )
+                }.first().key
+            )
+        }
+        fun downloadData(): ArchitecturyVersion? {
+            try {
+                val url1 = URL("https://maven.architectury.dev/dev/architectury/architectury/maven-metadata.xml")
+                val url2 = URL("https://maven.architectury.dev/me/shedaniel/architectury/maven-metadata.xml")
+                val result = mutableMapOf<SemanticVersion, MutableList<SemanticVersion>>()
+                url1.openStream().use { stream ->
+                    val inputFactory = XMLInputFactory.newInstance()
+
+                    @Suppress("UNCHECKED_CAST")
+                    val reader = inputFactory.createXMLEventReader(stream) as Iterator<XMLEvent>
+                    for (event in reader) {
+                        if (!event.isStartElement) {
+                            continue
+                        }
+                        val start = event.asStartElement()
+                        val name = start.name.localPart
+                        if (name != "version") {
+                            continue
+                        }
+
+                        val versionEvent = reader.next()
+                        if (!versionEvent.isCharacters) {
+                            continue
+                        }
+                        val version = versionEvent.asCharacters().data
+                        if (result.containsKey(findMcVersion(SemanticVersion.parse(version)))) {
+                            result[findMcVersion(SemanticVersion.parse(version))]!!.add(SemanticVersion.parse(version))
+                        } else {
+                            result[findMcVersion(SemanticVersion.parse(version))] = mutableListOf(
+                                SemanticVersion.parse(version)
+                            )
+                        }
+                    }
+                }
+                url2.openStream().use { stream ->
+                    val inputFactory = XMLInputFactory.newInstance()
```

Comment:
This code is a duplicate of the code above.

Reply:
there are 2 urls and they both need the same code.
i guess i could use a loop?

Reply:
nah i reworked it and made it better see below